message-expiration/message-expiration.bal

```diff
@@ -0,0 +1,37 @@
+import ballerina/http;
+import ballerina/mime;
+import ballerina/time;
+
+type OAuthRequest record {
+    string initializedTime;
+    decimal expiration;
+};
+
+string? initializedTime = ();
+decimal expiration = 0;
+
+service /storage on new http:Listener(8080) {
+    resource function post setUp(OAuthRequest req) {
+        initializedTime = req.initializedTime;
+        expiration = req.expiration;
+    }
+
+    resource function get image(string fileName) returns http:Response|error? {
+        string? createdTime = initializedTime;
+        if createdTime is string {
+            boolean isExpired = check validateTimeExpiration(createdTime, expiration);
```

Comment:
What if we send initialized time in the request as a header. expiration time can be a header value or one which is hardcoded one. 

Then we implement a http request interceptor for this validation. If the validation is successful we forward the request to it's resource. Else we return a http timeout response.

You can find a BBE from [here](https://ballerina.io/learn/by-example/http-request-interceptor/).

Reply:
wdyt?

Reply:
It seems good example to me. I will update that

Reply:
@manuranga wdyt?

Reply:
Let's not do for interceptor. I am not 100% sure but I think recommended practice is not to prefer interceptor.

Reply:
Can you please verify with @shafreenAnfar